test/lib/server/services/log/LogService.test.js

```diff
@@ -47,4 +50,12 @@ module.exports = () => {
         expect(logsForFill).to.lengthOf(3);
         expect(logsForFill.map(({ id }) => id)).to.eql([1, 3, 4]);
     });
+
+    it('Should successfully return the amount of root logs based off the limit', async () => {
+        expect(rootLogs.logs).to.lengthOf(10);
+    });
+
+    it('Should successfully return only root logs', async () => {
+        rootLogs.logs.forEach((log) => expect(log.parentLogId).to.equal(log.id));
+    });
```

Comment:
These can be merged in one. What you can do however, is that the pagination and offset works fine.